### DISCUSSION TITLE
Questions on Queries and Mutations

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [TanStack/query#5045] [Q&A] [answered]
On https://react-query-v3.tanstack.com/guides/queries, what does the `fetchTodoList` parameter do?

On https://react-query-v3.tanstack.com/guides/mutations, why is there an `axios.post('/todos', newTodo)` there? Is react query only for GET requests and not for POST, PUT, and PATCH requests?

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

> On https://react-query-v3.tanstack.com/guides/queries, what does the fetchTodoList parameter do?

It's the query function and is described above the example:

A function that returns a promise that:
- Resolves the data, or
- Throws an error

> On https://react-query-v3.tanstack.com/guides/mutations, why is there an axios.post('/todos', newTodo) there? Is react query only for GET requests and not for POST, PUT, and PATCH requests?

It's a POST request because it's a mutation that the POST verb suits best. TanStack Query can be used for any HTTP verb; it's agnostic of the actual network layer. Queries are commonly GET verbs and mutations are commonly anything else, although there's no hard rules 🙂
